### PR TITLE
refactor: application board entries

### PR DESCRIPTION
**Pull request by hanszhoou**
## Description

image

image

Closes # (NO_ISSUE)

https://mask.atlassian.net/browse/MF-410
https://mask.atlassian.net/browse/MF-613
https://mask.atlassian.net/browse/MF-611

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

File: packages/mask/src/social-network/ui.ts [hunk 1]
```diff
@@ -86,7 +86,10 @@ export async function activateSocialNetworkUIInner(ui_deferred: SocialNetworkUI.
     startPostListener()
     ui.collecting.currentVisitingIdentityProvider?.start(signal)
     ui.injection.pageInspector?.(signal)
-    if (Flags.toolbox_enabled) ui.injection.toolbox?.(signal)
+    if (Flags.toolbox_enabled) {
+        ui.injection.toolbox?.(signal, 'wallet')
```
Comment (Jack-Works) [inline packages/mask/src/social-network/ui.ts:90]: why not start both in the toolbox implementation?

Comment (zhouhanseng) [reply packages/mask/src/social-network/ui.ts:90]: injectToolboxAtFacebook

Comment (hanszhoou) [reply packages/mask/src/social-network/ui.ts:90]: Toolbox implementations for SNS have already included `injectToolboxHintAtMinds`, `injectToolboxAtFacebook` and `injectToolboxHintAtTwitter`, and maybe more in the future. I want to keep dry.

**Review by guanbinrui — DISMISSED** [review]

**Review by UncleBill — DISMISSED** [review]

**Review by Jack-Works — DISMISSED** [review]

**Review by Jack-Works — DISMISSED** [review]

**Review by guanbinrui — DISMISSED** [review]

**Comment by zhouhanseng** [comment]
@DimensionDev/qa

**Comment by hanszhoou** [comment]
@DimensionDev/qa

File: packages/mask/src/components/CompositionDialog/CompositionUI.tsx [hunk 2]
```diff
@@ -156,6 +156,11 @@ export const CompositionDialogUI = forwardRef<CompositionRef, CompositionProps>(
     const onSubmit = useCallback(() => {
         if (!Editor.current) return
         setSending(true)
+        console.log({
+            editor: Editor.current.value,
+            encryptionKind,
+            onSubmit: props.onSubmit,
+        })
```
Comment (guanbinrui) [inline packages/mask/src/components/CompositionDialog/CompositionUI.tsx]:
```suggestion
```

File: packages/mask/src/components/CompositionDialog/useSubmit.ts [hunk 3]
```diff
@@ -17,7 +17,7 @@ export function useSubmit(onClose: () => void, reason: 'timeline' | 'popup' | 'r
     return useCallback(
         async (info: SubmitComposition) => {
             const { content, encode, target } = info
-
+            console.log({ whoAmI, profiles: globalUIState.profiles })
```
Comment (guanbinrui) [inline packages/mask/src/components/CompositionDialog/useSubmit.ts]:
```suggestion
```

File: packages/mask/src/components/DataSource/usePersonaConnectStatus.ts [hunk 4]
```diff
@@ -38,3 +41,33 @@ export function usePersonaConnectStatus() {
...
+    const lastRecognized = useLastRecognizedIdentity()
+    const username = useMemo(() => {
+        return lastState.username || lastRecognized.identifier?.userId
+    }, [lastState, lastRecognized])
+    const personas = useMyPersonas()
+
+    const checkSNSConnectToCurrentPersona = useCallback((persona: PersonaInformation) => {
+        if (!username) return undefined
+        return persona.linkedProfiles.some((x) => x.identifier.userId === username)
+    }, [])
```
Comment (guanbinrui) [inline packages/mask/src/components/DataSource/usePersonaConnectStatus.ts]: Please make sure it's an empty deps.

File: packages/mask/src/components/DataSource/usePersonaConnectStatus.ts [hunk 5]
```diff
@@ -38,3 +41,33 @@ export function usePersonaConnectStatus() {
...
+    const lastState = useSetupGuideStatusState()
+    const { currentPersona } = PersonaContext.useContainer()
+    const lastRecognized = useLastRecognizedIdentity()
+    const username = useMemo(() => {
+        return lastState.username || lastRecognized.identifier?.userId
+    }, [lastState, lastRecognized])
+    const personas = useMyPersonas()
+
+    const checkSNSConnectToCurrentPersona = useCallback((persona: PersonaInformation) => {
+        if (!username) return undefined
```
Comment (guanbinrui) [inline packages/mask/src/components/DataSource/usePersonaConnectStatus.ts]: It should return `boolean`.

File: packages/mask/src/components/shared/ApplicationBoard.tsx [hunk 6]
```diff
@@ -142,26 +150,48 @@ function ApplicationBoardContent() {
                 .filter((x) => Boolean(x.entry.RenderEntryComponent)),
         [snsAdaptorPlugins, currentWeb3Network, chainId, account],
     )
-    const listedAppList = applicationList.filter((x) => !getUnlistedApp(x))
+
+    const [isPlaying, setPlaying] = useState(true)
```
Comment (guanbinrui) [inline packages/mask/src/components/shared/ApplicationBoard.tsx]:
```suggestion
    const [isPlaying, setIsPlaying] = useState(true)
```

File: packages/mask/src/components/shared/ApplicationBoard.tsx [hunk 7]
```diff
@@ -171,91 +201,138 @@ function ApplicationBoardContent() {
...
+            : ApplicationEntryStatus.shouldVerifyNextId
+            ? verifyPersona
+            : undefined
+    // #endregion
+
+    // #region tooltip hint
+    const tooltipHint = useMemo(() => {
+        if (application.isWalletConnectedRequired) return t('application_tooltip_hint_connect_wallet')
+        if (application.isWalletConnectedEVMRequired) return t('application_tooltip_hint_switch_to_evm_wallet')
+        if (!application.entry.nextIdRequired) return undefined
```
Comment (guanbinrui) [inline packages/mask/src/components/shared/ApplicationBoard.tsx]:
```suggestion
        if (!application.entry.nextIdRequired) return
```

File: packages/mask/src/components/shared/ApplicationBoard.tsx [hunk 8]
```diff
@@ -142,26 +150,48 @@ function ApplicationBoardContent() {
...
+                        {recommendFeatureAppList.map((application, i) => (
+                            <Slide index={i} key={i}>
+                                <RenderEntryComponent
+                                    key={application.entry.ApplicationEntryID}
+                                    application={application}
+                                />
+                            </Slide>
+                        ))}
+                    </Slider>
+                </CarouselProvider>
```
Comment (guanbinrui) [inline packages/mask/src/components/shared/ApplicationBoard.tsx]: Move to a new component file.

File: packages/mask/package.json [hunk 9]
```diff
@@ -97,6 +97,7 @@
     "millify": "^4.0.0",
     "next-tick": "^1.0.0",
     "opensea-js": "^1.2.7",
+    "pure-react-carousel": "^1.28.1",
```
Comment (Jack-Works) [inline packages/mask/package.json:100]: Nobody likes the carousel. https://shouldiuseacarousel.com/

Please recheck the design.

File: packages/plugins/GoPlusSecurity/src/SNSAdaptor/index.tsx [hunk 10]
```diff
@@ -12,7 +12,7 @@ const sns: Plugin.SNSAdaptor.Definition = {
     ApplicationEntries: [
         (() => {
             const icon = <SecurityCheckerIcon />
-            const name = { i18nKey: '__plugin_name', fallback: 'Check Security' }
```
Comment (Jack-Works) [inline packages/plugins/GoPlusSecurity/src/SNSAdaptor/index.tsx]: Why you're removing those keys? Please revert it.

File: packages/mask/src/components/DataSource/usePersonaConnectStatus.ts [hunk 11]
```diff
@@ -38,3 +41,34 @@ export function usePersonaConnectStatus() {
         return { connected, action, hasPersona: !!personas.length, currentConnectedPersona }
     }, [personas, lastRecognized, activatedSocialNetworkUI])
 }
+
+export function usePersonaAgainstSNSConnectStatus() {
```
Comment (Jack-Works) [inline packages/mask/src/components/DataSource/usePersonaConnectStatus.ts]: This hook looks overcomplicated can you simplify it? And please move it to its own file.

File: packages/mask/src/components/shared/ApplicationBoard.tsx [hunk 12]
```diff
@@ -171,91 +175,134 @@ function ApplicationBoardContent() {
...
-    const onNextIdVerify = useCallback(() => {
-        closeDialog()
+    const verifyPersona = useCallback(() => {
+        closeApplicationBoard()
         CrossIsolationMessages.events.verifyNextID.sendToAll(undefined)
     }, [])
 
-    if (!application.entry.RenderEntryComponent) return null
+    const clickHandler =
+        application.isWalletConnectedRequired || application.isWalletConnectedEVMRequired
```
Comment (Jack-Works) [inline packages/mask/src/components/shared/ApplicationBoard.tsx]: Please simplify this expression.

File: packages/mask/src/components/shared/ApplicationBoard.tsx [hunk 13]
```diff
@@ -171,91 +175,134 @@ function ApplicationBoardContent() {
...
+            ? undefined
+            : ApplicationEntryStatus.isPersonaConnected === false || ApplicationEntryStatus.isPersonaCreated === false
+            ? createOrConnectPersona
+            : ApplicationEntryStatus.shouldVerifyNextId
+            ? verifyPersona
+            : undefined
+    // #endregion
+
+    // #region tooltip hint
+    const tooltipHint = useMemo(() => {
```
Comment (Jack-Works) [inline packages/mask/src/components/shared/ApplicationBoard.tsx]: I think you don't need to memo this 🤔

File: packages/mask/package.json (hunk 9 above)
Comment (hanszhoou) [reply packages/mask/package.json:100]: PM was requiring carousel.

File: packages/mask/src/components/shared/ApplicationBoard.tsx (hunk 13 above)
Comment (hanszhoou) [reply packages/mask/src/components/shared/ApplicationBoard.tsx]: 🤔

File: packages/mask/src/components/DataSource/usePersonaConnectStatus.ts (hunk 11 above)
Comment (hanszhoou) [reply packages/mask/src/components/DataSource/usePersonaConnectStatus.ts]: Moved, but don't know how to simplify it now.

File: packages/plugins/GoPlusSecurity/src/SNSAdaptor/index.tsx (hunk 10 above)
Comment (hanszhoou) [reply packages/plugins/GoPlusSecurity/src/SNSAdaptor/index.tsx]: done

File: packages/mask/src/components/shared/ApplicationBoard.tsx (hunk 12 above)
Comment (hanszhoou) [reply packages/mask/src/components/shared/ApplicationBoard.tsx]: done

File: packages/mask/src/components/shared/ApplicationBoard.tsx (hunk 13 above)
Comment (hanszhoou) [reply packages/mask/src/components/shared/ApplicationBoard.tsx]: done

File: packages/mask/package.json (hunk 9 above)
Comment (Jack-Works) [reply packages/mask/package.json:100]: I'm not blocking this, just want to let them know

File: packages/shared/src/contexts/components/InjectedDialog.tsx [hunk 14]
```diff
@@ -74,6 +75,7 @@ export function InjectedDialog(props: InjectedDialogProps) {
         dialogCloseButton,
         dialogContent,
         dialogTitle,
+        dialogTitleTail,
```
Comment (Jack-Works) [inline packages/shared/src/contexts/components/InjectedDialog.tsx]: Looks like this new class is unused.

File: packages/plugins/CrossChainBridge/src/SNSAdaptor/index.tsx [hunk 15]
```diff
@@ -12,7 +12,7 @@ const sns: Plugin.SNSAdaptor.Definition = {
     ApplicationEntries: [
         (() => {
             const icon = <CrossBridgeIcon />
-            const name = { i18nKey: '__plugin_name', fallback: 'Cross-bridge' }
+            const name = { fallback: 'Cross-bridge' }
```
Comment (Jack-Works) [inline packages/plugins/CrossChainBridge/src/SNSAdaptor/index.tsx]: Revert this change plz.

File: packages/mask/src/components/shared/ApplicationRecommendArea.tsx [hunk 16]
```diff
@@ -0,0 +1,77 @@
...
+                    visibleSlides={2.2}
+                    infinite={false}
+                    interval={2500}
+                    className={classes.carousel}
+                    isPlaying={isPlaying}>
+                    <Slider onScroll={(e) => setIsPlaying((e.target as HTMLDivElement).scrollLeft === 0)}>
+                        {recommendFeatureAppList.map((application, i) => (
+                            <Slide index={i} key={i}>
+                                <RenderEntryComponent
+                                    key={application.entry.ApplicationEntryID}
```
Comment (Jack-Works) [inline packages/mask/src/components/shared/ApplicationRecommendArea.tsx]: unnecessary key.

File: packages/mask/src/components/shared/ApplicationRecommendArea.tsx [hunk 17]
```diff
@@ -0,0 +1,77 @@
...
+                    naturalSlideHeight={117}
+                    totalSlides={recommendFeatureAppList.length}
+                    visibleSlides={2.2}
+                    infinite={false}
+                    interval={2500}
+                    className={classes.carousel}
+                    isPlaying={isPlaying}>
+                    <Slider onScroll={(e) => setIsPlaying((e.target as HTMLDivElement).scrollLeft === 0)}>
+                        {recommendFeatureAppList.map((application, i) => (
+                            <Slide index={i} key={i}>
```
Comment (Jack-Works) [inline packages/mask/src/components/shared/ApplicationRecommendArea.tsx]: should be `application.entry.ApplicationEntryID`.

File: packages/mask/src/components/DataSource/usePersonaAgainstSNSConnectStatus.ts [hunk 18]
```diff
@@ -0,0 +1,40 @@
...
+import { useLastRecognizedIdentity } from './useActivatedUI'
+import { useMyPersonas } from './useMyPersonas'
+import { useSetupGuideStatusState } from './useNextID'
+import { PersonaContext } from '../../extension/popups/pages/Personas/hooks/usePersonaContext'
+
+export function usePersonaAgainstSNSConnectStatus() {
+    const ui = activatedSocialNetworkUI
+    const platform = ui.configuration.nextIDConfig?.platform as NextIDPlatform
+    const lastState = useSetupGuideStatusState()
+    const { currentPersona } = PersonaContext.useContainer()
```
Comment (Jack-Works) [inline packages/mask/src/components/DataSource/usePersonaAgainstSNSConnectStatus.ts]: I'd prefer avoid using PersonaContext here. It is for popups.

Comment (Jack-Works) [reply packages/mask/src/components/DataSource/usePersonaAgainstSNSConnectStatus.ts]: It looks like you're only using it for deps array. Please change to listen to the persona update event and re-trigger the async function.

File: packages/mask/src/components/DataSource/usePersonaAgainstSNSConnectStatus.ts [hunk 19]
```diff
@@ -0,0 +1,40 @@
...
+    const lastRecognized = useLastRecognizedIdentity()
+    const username = useMemo(() => {
+        return lastState.username || lastRecognized.identifier?.userId
+    }, [lastState, lastRecognized])
+    const personas = useMyPersonas()
+
+    const checkSNSConnectToCurrentPersona = useCallback(
```
Comment (Jack-Works) [inline packages/mask/src/components/DataSource/usePersonaAgainstSNSConnectStatus.ts]: This function is only used in the async function. Why you need to memo it? Please move it into the async function.

**Review by Jack-Works — DISMISSED** [review]
request changes

File: packages/mask/src/components/DataSource/usePersonaAgainstSNSConnectStatus.ts [hunk 20]
```diff
@@ -0,0 +1,34 @@
...
+
+export function usePersonaAgainstSNSConnectStatus() {
+    const ui = activatedSocialNetworkUI
+    const platform = ui.configuration.nextIDConfig?.platform as NextIDPlatform
+    const lastState = useSetupGuideStatusState()
+    const lastRecognized = useLastRecognizedIdentity()
+    const username = useMemo(() => {
+        return lastState.username || lastRecognized.identifier?.userId
+    }, [lastState, lastRecognized])
+    const personas = useMyPersonas()
```
Comment (Jack-Works) [inline packages/mask/src/components/DataSource/usePersonaAgainstSNSConnectStatus.ts:18]: looks like all of those data are only used in the async function. I think you should move the async function to the top level and subscribe to the events to upgrade/retry the async function.

Comment (hanszhoou) [reply packages/mask/src/components/DataSource/usePersonaAgainstSNSConnectStatus.ts:18]: Why not use those data directly 🤔

**Review by BillySh1 — APPROVED** [review]

**Review by lelenei — APPROVED** [review]
